src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+            {
+                if (e.RequestInformation?.HttpStatusCode == (int)HttpStatusCode.NotFound)
+                {
+                    errorMessage = $"Lease file data could not be found for blob on Partition: '{partitionId}', " +
+                        $"EventHub: '{_eventHubName}', '{_consumerGroup}'. Error: {e.Message}";
+                }
+            }
+            catch (JsonSerializationException e)
+            {
+                errorMessage = $"Could not deserialize blob lease info for blob on Partition: '{partitionId}', " +
+                    $"EventHub: '{_eventHubName}', Consumer Group: '{_consumerGroup}'. Error: {e.Message}";
+            }
+            catch (Exception e)
+            {
+                errorMessage = $"Encountered exception while checking for last checkpointed sequence number for blob " +
+                    $"on Partition: '{partitionId}', EventHub: '{_eventHubName}', Consumer Group: '{_consumerGroup}'. Error: {e.Message}";
+            }
+
+            return new Tuple<BlobPartitionLease, string>(blobPartitionLease, errorMessage);
+        }
+
+        // Get the number of unprocessed events by deriving the delta between the server side info and the partition lease info, 
+        private long GetUnprocessedEventCount(EventHubPartitionRuntimeInformation partitionInfo, BlobPartitionLease partitionLeaseInfo)
+        {
+            long partitionLeaseInfoSequenceNumber = partitionLeaseInfo.SequenceNumber ?? 0;
+
+            // This handles two scenarios:
+            //   1. If the partition has received its first message, Offset will be null and LastEnqueuedSequenceNumber will be 0
+            //   2. If there are no instances set to process messages, Offset will be null and LastEnqueuedSequenceNumber will be >= 0
+            if (partitionLeaseInfo.Offset == null && partitionInfo.LastEnqueuedSequenceNumber >= 0)
+            {
+                return (partitionInfo.LastEnqueuedSequenceNumber + 1);
+            }
+
+            if (partitionInfo.LastEnqueuedSequenceNumber > partitionLeaseInfoSequenceNumber)
+            {
+                return (partitionInfo.LastEnqueuedSequenceNumber - partitionLeaseInfoSequenceNumber);
+            }
+
+            // Partition is a circular buffer, so it is possible that
+            // LastEnqueuedSequenceNumber < SequenceNumber
+            long count = 0;
+            unchecked
+            {
+                count = (long.MaxValue - partitionInfo.LastEnqueuedSequenceNumber) + partitionLeaseInfoSequenceNumber;
+            }
+
+            // It's possible for checkpointing to be ahead of the partition's LastEnqueuedSequenceNumber,
+            // especially if checkpointing if happening often and load is very low.
+            // If count is negative, we need to know that this read is invalid, so return 0.
+            // e.g., (9223372036854775807 - 10) + 11 = -9223372036854775808
+            return (count < 0) ? 0 : count;

```

Comment:
Ok so I may be missing something really obvious, but is there a reason we don't just put an `else { return 0; }` here instead of the unchecked arithmetic? I can't think of a situation when count would be positive...

Reply:
Yeah, not sure. Let's assume for now they had a good reason. We can refactor in the future as necessary